compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -677,7 +677,9 @@ trait Applications extends Compatibility {
    */
   class ApplicableToTrees(methRef: TermRef, args: List[Tree], resultType: Type)(using Context)
   extends TestApplication(methRef, methRef.widen, args, resultType) {
-    def argType(arg: Tree, formal: Type): Type = normalize(arg.tpe, formal)
+    def argType(arg: Tree, formal: Type): Type =
+      if untpd.isContextualClosure(arg) && defn.isContextFunctionType(formal) then arg.tpe
+      else normalize(arg.tpe, formal)
```

Comment:
Can we instead check whether subtype conforms? The following code fails to type-check:

```Scala
def foo(init: String ?=> Int ?=> Int) : Int = 1
def foo(s: String)(init: Int ?=> Int) : Int = 2


@main def Test() =
  implicit val ev: String = "hello"
  assert(foo((using x:Int) => x) == 1)
```

Reply:
I think that goes beyond what's possible. That code also fails if only the first variant of `foo` is given. 

I don't see any way to improve the current solution.

Reply:
Yes, I see the eta-expansion is before type checking the argument, it's pretty subtle.